subsys: random: Fix semaphore initial count of xoroshiro128

ramakrishnapallala opened the pull request:
Fix semaphore initial count of rand32_xoroshiro128

Signed-off-by: Ramakrishna Pallala <ramakrishna.pallala@intel.com>

galak reviewed (changes requested):
Can you explain why in the commit message this change is needed.

ramakrishnapallala commented:
in this driver, originally the semaphore initial count is set to 0 but k_sem_give() is invoked first and the caller will be blocked forever.

please check sys_rand32_get() in subsys/random/rand32_xoroshiro128.c

galak commented:
> in this driver, originally the semaphore initial count is set to 0 but k_sem_give() is invoked first and the caller will be blocked forever.
> 
> please check sys_rand32_get() in subsys/random/rand32_xoroshiro128.c

can you add this to the commit message

Commit:
subsys: random: Fix semaphore initial count of xoroshiro128

Semaphore state_sem initial count is set to 0 but k_sem_give()
is invoked first in sys_rand32_get() which will block the caller
forever.

Fix the issue by setting the initail count to count_limit.

Signed-off-by: Ramakrishna Pallala <ramakrishna.pallala@intel.com>

lpereira reviewed (approved):
LGTM. Thanks for spotting and fixing this.

andyross reviewed (approved):
Seems obvious
